# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalChild.h
# Commit: 405100db6c45
# Full Hash: 405100db6c45f0463286067df5b491f74790d9aa
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-11 09:51:12
# Regressor Bug: 1543251
# File Overlap Count: 1
# Description:
#   Bug 1543251 - Move hasBeforeUnload from PBrowser to PWindowGlobal, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37003
# ==============================================================================

diff -r 0c408343080d -r 405100db6c45 dom/ipc/WindowGlobalChild.h
--- a/dom/ipc/WindowGlobalChild.h	Thu Jul 04 07:41:57 2019 +0000
+++ b/dom/ipc/WindowGlobalChild.h	Wed Jul 10 21:13:44 2019 +0000
@@ -63,6 +63,10 @@
 
   uint64_t ContentParentId();
 
+  int64_t BeforeUnloadListeners() { return mBeforeUnloadListeners; }
+  void BeforeUnloadAdded();
+  void BeforeUnloadRemoved();
+
   bool IsCurrentGlobal();
 
   bool IsProcessRoot();
@@ -122,6 +126,7 @@
   nsRefPtrHashtable<nsStringHashKey, JSWindowActorChild> mWindowActors;
   uint64_t mInnerWindowId;
   uint64_t mOuterWindowId;
+  int64_t mBeforeUnloadListeners;
   bool mIPCClosed;
 };
 